openvdb/openvdb/points/PointRasterizeSDF.h

```diff
@@ -0,0 +1,1701 @@
+/// @details There are two main transfer implementations; rasterizeSpheres and
+///  rasterizeSmoothSpheres. The prior performs trivial narrow band stamping
+///  of spheres for each point, where as the latter calculates an averaged
+///  position of influence per voxel as described in:
+///    [Animating Sand as a Fluid - Zhu Bridson 05].
+///
+///  rasterizeSpheres() is an extremely fast and efficient way to produce both a
+///  valid symmetrical narrow band level set and transfer attributes using
+///  closest point lookups.
+///
+///  rasterizeSmoothSpheres() produces smoother, more blended connections
+///  between points which is ideal for generating a more artistically pleasant
+///  surface directly from point distributions. It aims to avoid typical post
+///  filtering operations used to smooth surface volumes. Note however that
+///  rasterizeSmoothSpheres may not necessarily produce a *symmetrical* narrow
+///  band level set; the exterior band may be smaller than desired depending on
+///  the search radius. The surface can be rebuilt or resized if necessary.
+///  The same closet point algorithm is used to transfer attributes.
+///
+///  In general, it is recommended to consider post rebuilding/renormalizing the
+///  generated surface using either tools::levelSetRebuild() or
+///  tools::LevelSetTracker::normalize() tools::LevelSetTracker::resize().
+///
+/// @note These methods use the framework provided in PointTransfer.h
+///
+
+#ifndef OPENVEB_POINTS_RASTERIZE_SDF_HAS_BEEN_INCLUDED
+#define OPENVEB_POINTS_RASTERIZE_SDF_HAS_BEEN_INCLUDED
+
+#include "PointDataGrid.h"
+#include "PointTransfer.h"
+#include "PointStatistics.h"
+
+#include <openvdb/openvdb.h>
+#include <openvdb/Types.h>
+#include <openvdb/tools/Prune.h>
+#include <openvdb/tools/ValueTransformer.h>
+#include <openvdb/thread/Threading.h>
+#include <openvdb/util/NullInterrupter.h>
+
+#include <unordered_map>
+
+#include <tbb/task_group.h>
+#include <tbb/parallel_reduce.h>
+
+namespace openvdb {
+OPENVDB_USE_VERSION_NAMESPACE
+namespace OPENVDB_VERSION_NAME {
+namespace points {
+
+/// @brief Narrow band sphere stamping with a uniform radius.
+/// @details Rasterizes points into a level set using basic sphere stamping with
+///   a uniform radius. The radius parameter is given in world space units and
+///   is applied to every point to generate a fixed surface mask and consequent
+///   distance values.
+/// @param points       the point data grid to rasterize
+/// @param radius       the world space radius of every point
+/// @param halfband     the half band width
+/// @param transform    the target transform for the surface
+/// @param filter       a filter to apply to points
+/// @param interrupter  optional interrupter
+/// @return The signed distance field.
+template <typename PointDataGridT,
+    typename SdfT = typename PointDataGridT::template ValueConverter<float>::Type,
+    typename FilterT = NullFilter,
+    typename InterrupterT = util::NullInterrupter>
```

Comment:
The interrupter should not be provided as a template argument. This prevents us doing explicit template instantiation. There are lots of old methods that still do this that we haven't changed, but for new methods, it should really just be part of the function signature (ie `util::NullInterrupter* interrupter = nullptr`).

Reply:
Whilst I agree, we can't do explicit template instantiation for this anyway as it takes a `FilterT` :(